brainiak/matnormal/covs.py

```diff
@@ -0,0 +1,542 @@
+import tensorflow as tf
+import numpy as np
+import abc
+import scipy.linalg
+import scipy.sparse
+from tensorflow.contrib.distributions import InverseGamma, WishartCholesky
+from brainiak.matnormal.utils import define_scope, xx_t
+from brainiak.utils.utils import tf_solve_lower_triangular_kron,\
+                          tf_solve_upper_triangular_kron, \
+                          tf_solve_lower_triangular_masked_kron, \
+                          tf_solve_upper_triangular_masked_kron
+
+__all__ = ['CovBase',
+           'CovIdentity',
+           'CovAR1',
+           'CovIsotropic',
+           'CovDiagonal',
+           'CovDiagonalGammaPrior',
+           'CovUnconstrainedCholesky',
+           'CovUnconstrainedCholeskyWishartReg',
+           'CovUnconstrainedInvCholesky',
+           'CovKroneckerFactored']
+
+
+class CovBase:
+    """Base metaclass for noise covariances
+    """
```

Comment:
As per @lcnature, I should provide a little more info about how ABCs work and how devs should subclass from it, etc.

Reply:
Because we are using Python 3, you should subclass from `abc.ABC`.

Regarding providing more info, a note that the class has abstract methods with a link to the Python library docs (https://docs.python.org/3/library/abc.html#abc.abstractmethod) should be enough.

Reply:
done.